tests/unit/test_runner.py

```diff
@@ -416,3 +431,30 @@ def test_remove_with_delete_error(
 
     with pytest.raises(RunnerRemoveError):
         runner.remove("test_token")
+
+
+def test_random_ssh_connection_choice(
+    runner: Runner,
+    vm_resources: VirtualMachineResources,
+    token: str,
+    binary_path: Path,
+):
+    """
+    arrange: given a mock runner with random batch of ssh debug infos.
+    act: when runner.configure_runner is called.
+    assert: selected ssh_debug_info is random.
+    """
+    runner.create(
+        config=CreateRunnerConfig(
+            image="test_image",
+            resources=vm_resources,
+            binary_path=binary_path,
+            registration_token=token,
+        )
+    )
+    runner._configure_runner()
+    first_call_args = runner._clients.jinja.get_template("env.j2").render.call_args.kwargs
+    runner._configure_runner()
+    second_call_args = runner._clients.jinja.get_template("env.j2").render.call_args.kwargs
+
+    assert first_call_args["ssh_debug_info"] != second_call_args["ssh_debug_info"]
```

Comment:
Perhaps add a message that this is unexpected, but could fail, but with a low probability, and can be retried.